### ISSUE TITLE
Token program instruction builders dont support token2022

### ISSUE THREAD

**Issue by Jac0xb** [febo/pinocchio#39]
### Issue

- Token programs have hard coded legacy token program in the instruction builders.

**Comment by sonicfromnewyoke** [comment]
I think it is ok, and there will be another lib `token-2022` with almost a copy-paste code from the legacy token program with hardcoded `TokenzQdBNbLqP5VEhdkAS6EPFLC1PHnBqCXEpPxuEb`.

The same approach was used in fndn

**Comment by Jac0xb** [comment]
I do not hold the original libs in high regards. This seems like a poor use of space if I am including these in my onchain program. Almost all the instructions are identical and there could easily be a field in the struct for token program rather than needed to depend on two packages and have conditional logic that duplicates the code for each version.

**Comment by sonicfromnewyoke** [comment]
> I do not hold the original libs in high regards

makes sense

**Comment by Nagaprasadvr** [comment]
Makes sense but this would increase 32 bytes for each common ix wyt @febo

**Comment by febo** [comment]
Since `token-2022` is a superset of the token instructions, perhaps the `token-2022` client should include the token program as a parameter. This way it can be used for both `token` and `token-2022`.

**Comment by Nagaprasadvr** [comment]
so we just move all token ixs to token-2022 client  and add an enum field to mark wither legacy or token-2022 ig

**Comment by febo** [comment]
> so we just move all token ixs to token-2022 client and add an enum field to mark whether legacy or token-2022 ig

Not sure if we need an `enum` field. We could have a field for the token program's `AccountInfo` and use the `key` from there.

**Comment by Taylor123** [comment]
Has any more thought been put into Token2022 support? Would be great to have the token crate support both.

**Comment by Nagaprasadvr** [comment]
> Has any more thought been put into Token2022 support? Would be great to have the token crate support both.

I have a pr tat i was working on https://github.com/anza-xyz/pinocchio/pull/36

**Comment by alizeeshan1234** [comment]
[pinocchioPreReq1.docx](https://github.com/user-attachments/files/21009023/pinocchioPreReq1.docx)

**Comment by febo** [comment]
Support was added in PR #190
